### PR TITLE
[explore-v2] add fave star and edit button to chart header

### PR DESCRIPTION
**Pull request by ascott**
screenshot 2016-11-16 22 33 07

plz review @vera-liu @mistercrunch @bkyryliuk

### CODE REVIEW

**Comment by mistercrunch** [comment]
This PR needs to be linted to pass the build

File: superset/assets/javascripts/components/FaveStar.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  isStarred: PropTypes.bool.isRequired,
+};
+
+export default class FaveStar extends React.Component {
+  componentDidMount() {
+    this.props.actions.fetchFaveStar(this.props.sliceId);
+  }
+
+  onClick(e) {
+    e.preventDefault();
```
Comment (bkyryliuk) [inline superset/assets/javascripts/components/FaveStar.jsx]: I am curious, what is default behavior?

File: superset/assets/javascripts/components/TooltipWrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  children: PropTypes.node.isRequired,
+  placement: PropTypes.string,
+};
+
+const defaultProps = {
+  placement: 'top',
+}
+
+export default function TooltipWrapper({ label, tooltip, children, placement}) {
+  console.log('children', children)
```
Comment (bkyryliuk) [inline superset/assets/javascripts/components/TooltipWrapper.jsx]: leftovers ?

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 3]
```diff
@@ -49,6 +53,33 @@ export function fetchFieldOptions(datasourceId, datasourceType) {
...
 
+export const FETCH_FAVE_STAR = 'FETCH_FAVE_STAR'
+export function fetchFaveStar(sliceId) {
+  return function (dispatch) {
+    const url = `${FAVESTAR_BASE_URL}/${sliceId}/count`;
+    $.get(url, (data) => {
+      if (data.count > 0) {
+        dispatch(toggleFaveStar(true));
+      }
+    });
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: codeclimate says: Missing semicolon.

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 4]
```diff
@@ -49,6 +53,33 @@ export function fetchFieldOptions(datasourceId, datasourceType) {
   };
 }
 
+export const FETCH_FAVE_STAR = 'FETCH_FAVE_STAR'
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: codeclimate says: Missing semicolon.

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 5]
```diff
@@ -49,6 +53,33 @@ export function fetchFieldOptions(datasourceId, datasourceType) {
...
+export function fetchFaveStar(sliceId) {
+  return function (dispatch) {
+    const url = `${FAVESTAR_BASE_URL}/${sliceId}/count`;
+    $.get(url, (data) => {
+      if (data.count > 0) {
+        dispatch(toggleFaveStar(true));
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: codeclimate says: "toggleFaveStar" was used before it was defined

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 6]
```diff
@@ -49,6 +53,33 @@ export function fetchFieldOptions(datasourceId, datasourceType) {
...
+      }
+    });
+  }
+}
+
+export const SAVE_FAVE_STAR = 'SAVE_FAVE_STAR'
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: codeclimate says: Missing semicolon.

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 7]
```diff
@@ -49,6 +53,33 @@ export function fetchFieldOptions(datasourceId, datasourceType) {
...
+export function saveFaveStar(sliceId, isStarred) {
+  return function (dispatch) {
+    const urlSuffix = isStarred ? 'unselect' : 'select';
+    const url = `${FAVESTAR_BASE_URL}/${sliceId}/${urlSuffix}/`;
+    $.get(url);
+    dispatch(toggleFaveStar(!isStarred));
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: codeclimate says: "toggleFaveStar" was used before it was defined

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 8]
```diff
@@ -49,6 +53,33 @@ export function fetchFieldOptions(datasourceId, datasourceType) {
...
+  }
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: codeclimate says: Missing semicolon.

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 9]
```diff
@@ -1,6 +1,10 @@
 /* eslint camelcase: 0 */
 const $ = window.$ = require('jquery');
 
+// todo: fix bug in favstar endpoint that requires
+// that /undefined/ is part of this url.
+const FAVESTAR_BASE_URL = '/superset/favstar/undefined';
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: s/undefined/Slice: 
` @expose("/favstar/<class_name>/<obj_id>/<action>/")`

File: superset/assets/javascripts/explorev2/actions/exploreActions.js [hunk 10]
```diff
@@ -49,6 +53,33 @@ export function fetchFieldOptions(datasourceId, datasourceType) {
   };
 }
 
+export const FETCH_FAVE_STAR = 'FETCH_FAVE_STAR'
+export function fetchFaveStar(sliceId) {
+  return function (dispatch) {
+    const url = `${FAVESTAR_BASE_URL}/${sliceId}/count`;
```
Comment (bkyryliuk) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: there is no such action is count, I believe you can keep it with no action and the endpoint will return you the count

Comment (ascott) [inline superset/assets/javascripts/explorev2/actions/exploreActions.js]: 👍

File: superset/assets/javascripts/components/TooltipWrapper.jsx (hunk 2 above)
Comment (ascott) [inline superset/assets/javascripts/components/TooltipWrapper.jsx]: yep. looks like my linter stopped working in sublime for some reason, so i missed this. thanks for the :eyes:

**Pull request merged**